node/commons/compatibility-layer.ts

```diff
@@ -454,6 +454,19 @@ export const buildBreadcrumb = (
     pivotValue.push(value.key)
     pivotMap.push(value.attributeKey)
 
+    if (value.attributeKey === "productClusterIds") {
+      const clusterName = attributes.filter(attribute => attribute.key === "productclusternames")?.map(attribute => attribute.values)?.reduce((acc, val) => acc.concat(val), []).filter(attrValue => attrValue.id === value.key)
```

Comment:
Break this line into multiple lines, please. It is hard to read.

Why are you using `filter` to get the attribute with the  `productclusternames ` key?
Shouldn't it be `find`? There is more than one `productclustnames` attribute `?

It's not clear to me why the `clusterNames` is an array and not a string

Reply:
Good catch, I can use the find.